src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -187,9 +194,60 @@ public double getY(){
   MechanismLigament2d elbowLigament = baseLigament.append(new MechanismLigament2d("Elbow Arm", kElbowLength*3, elbowArmSim.getAngleRads()));
   
 
+  public ArmSubsystem(IntakeSubsystem intakeSubsystem) {
+    //Add coast mode command to shuffleboard
+    configTab.add(new StartEndCommand(this::coast, this::brake, this).ignoringDisable(true).withName("Coast Arm"));
+
+    baseMotor = MotorController.constructMotor(MotorConfig.ArmBase1);
+    baseMotor2 = MotorController.constructMotor(MotorConfig.ArmBase2);
+    elbowMotor = MotorController.constructMotor(MotorConfig.ArmElbow1);
+    elbowMotor2 = MotorController.constructMotor(MotorConfig.ArmElbow2);
+
+    baseMotor2.follow(baseMotor);
+    elbowMotor2.follow(elbowMotor);
+
+    baseMotor.enableVoltageCompensation(11);
+    elbowMotor.enableVoltageCompensation(11);
+
+    baseAbsoluteEncoder = AbsoluteEncoder.constructREVEncoder(EncoderConfig.ArmBase);
+    elbowAbsoluteEncoder = AbsoluteEncoder.constructREVEncoder(EncoderConfig.ArmElbow);
+    choochooAbsoluteEncoder = AbsoluteEncoder.constructREVEncoder(EncoderConfig.ArmChooChoo);
+    this.intakeSubsystem = intakeSubsystem;
+
+    if(Robot.isReal()) {
+      basePIDController = new ProfiledPIDController(kBaseP, kBaseI, kBaseD, kBaseConstraints);
+      elbowPIDController = new ProfiledPIDController(kElbowUpP, kElbowUpI, kElbowUpD, kElbowConstraints);
+      elbowFeedForward = new ArmFeedforward(kElbowS, kElbowG, kElbowV, kElbowA);
+      
+      basePTuner = new TunableNumber("baseP", kBaseP, basePIDController::setP);
+      baseITuner = new TunableNumber("baseI", kBaseI, basePIDController::setI);
+      baseDTuner = new TunableNumber("baseD", kBaseD, basePIDController::setD);
+  
+      elbowUpPTuner = new TunableNumber("elbowUpP", kElbowUpP, (a) -> kElbowUpP = a);
+      elbowUpITuner = new TunableNumber("elbowUpI", kElbowUpI, (a) -> kElbowUpI = a);
+      elbowUpDTuner = new TunableNumber("elbowUpD", kElbowUpD, (a) -> kElbowUpD = a);
+
+      elbowDownPTuner = new TunableNumber("elbowDownP", kElbowDownP, (a) -> kElbowDownP = a);
+      elbowDownITuner = new TunableNumber("elbowDownI", kElbowDownI, (a) -> kElbowDownI = a);
+      elbowDownDTuner = new TunableNumber("elbowDownD", kElbowDownD, (a) -> kElbowDownD = a);
+    } else {
+      SmartDashboard.putData("Arm Sim", simArmCanvas);
+      elbowFeedForward = new ArmFeedforward(0, 0, 0, 0);
+      basePIDController = new ProfiledPIDController(kSimBaseP, 0, 0, kBaseConstraints);
+      elbowPIDController = new ProfiledPIDController(kSimElbowP, 0, 0, kElbowConstraints);
+    }
+
+    basePIDController.reset(getBaseAngle());
+    elbowPIDController.reset(getElbowAngle());
+
+    holdCurrentPosition();
+    
+    setState(kDefaultState);
```

Comment:
Just use the other constructor to do the common parts. Don't copy and paste methods because it makes code less maintainable. 
```suggestion
    this.intakeSubsystem = intakeSubsystem;
    this();
```